lib/route_matcher.rb

```diff
@@ -54,7 +54,7 @@ def params_allowed?(request)
 
     params.all? do |param|
       param_alias = aliases&.[](param)
-      allowed_values = [allowed_param_values[param.to_s]].flatten
+      allowed_values = [allowed_param_values.fetch(param.to_s, [])].flatten
```

Comment:
You can also do `.compact.flatten` instead of adding empty arrays.

Reply:
Why is this change necessary?